src/Microsoft.AspNetCore.Authentication.Impl/AuthenticationServiceCollectionExtensions.cs

```diff
@@ -46,8 +45,7 @@ public static IServiceCollection AddAuthentication(this IServiceCollection servi
             return services;
         }
 
-        // REVIEW: rename to just AddScheme?
-        public static IServiceCollection AddSchemeHandler<TOptions, THandler>(this IServiceCollection services, string authenticationScheme, Action<TOptions> configureOptions)
+        public static IServiceCollection AddScheme<TOptions, THandler>(this IServiceCollection services, string authenticationScheme, Action<TOptions> configureOptions, bool canHandleRequests)
```

Comment:
This is the infrastructure method.  AddGoogle/Cookies/FacebookAuth use this method to register themselves and set the flag.  The idea was to make it opt in for the schemes to get called by the middleware, but as @Tratcher mentioned, next I'll probably scope the blanket callback the handlers get today, and instead they would register their callback paths and the middleware would do some ghetto routing, instead of this opt in bit..  You'd still be able to do more advanced routing by replacing the AuthenticationMiddleware (or adding another customer middleware).  The motivation is to avoid creating the handlers for all the schemes until they are needed/called.